protocols/request-response/src/lib.rs

```diff
@@ -437,8 +437,19 @@ where
     /// by [`NetworkBehaviour::handle_pending_outbound_connection`].
     ///
     /// Addresses added in this way are only removed by `remove_address`.
-    pub fn add_address(&mut self, peer: &PeerId, address: Multiaddr) {
-        self.addresses.entry(*peer).or_default().push(address);
+    ///
+    /// Returns true if the address was added, false otherwise (i.e. if the
+    /// address is already in the list).
+    pub fn add_address(&mut self, peer: &PeerId, address: Multiaddr) -> bool {
```

Comment:
I wonder if its work making this take `&Multiaddr`and then only cloning if we need to add it.

It'd mean the consumer of the API doesn't have to take ownership to use it (which may involve a pointless clone, as we have atm).

Reply:
Note that `Multiaddr` is a wrapper around an `Arc<Vec<_>>`. Thus I doubt additional clones have a performance impact here.

https://github.com/multiformats/rust-multiaddr/blob/d0e9ec219645ce8bd8ae7926e4b7316a67c35f84/src/lib.rs#L40-L45

Reply:
Ah nice one. Good to know! 🙇